ui/components/reservation/EditStep0.tsx

```diff
@@ -304,36 +304,38 @@ const EditStep0 = ({
 
   const handleEventChange = useCallback(
     (
-      { start, end }: CalendarEvent<ReservationType>,
+      { start, end }: CalendarEvent<Reservation | ReservationType>,
       skipLengthCheck = false
     ): boolean => {
-      const normalizedEnd = roundToNearestMinutes(end);
+      const { minReservationDuration, reservationStartInterval } =
+        reservationUnit;
 
-      const newReservation = {
+      const { end: minEnd } = getMinReservation({
+        begin: start,
+        minReservationDuration,
+        reservationStartInterval,
+      });
+
+      let normalizedEnd = getValidEndingTime({
+        start,
+        end: roundToNearestMinutes(end),
+        reservationStartInterval,
+      });
+
+      if (normalizedEnd < minEnd) {
+        normalizedEnd = minEnd;
+      }
```

Comment:
I think I've seen this block four times in this file and` ui/pages/reservation-unit/[id].tsx`
Is it the same? can it be moved to `common/calendar/utils`?